crates/wasmtime/src/component/component.rs

```diff
@@ -379,6 +388,28 @@ impl Component {
     pub(crate) fn runtime_info(&self) -> Arc<dyn ComponentRuntimeInfo> {
         self.inner.clone()
     }
+
+    pub(crate) fn resource_drop_func_ref(&self, dtor: &crate::func::HostFunc) -> VMFuncRef {
+        // Host functions never have their `wasm_call` filled in at this time.
+        assert!(dtor.func_ref().wasm_call.is_none());
+
+        // TODO
+        let wasm_call = self
+            .inner
+            .info
+            .resource_drop_wasm_to_native_trampoline
+            .as_ref()
+            .map(|i| {
+                let ptr = self.func(i);
+                unsafe {
+                    mem::transmute::<NonNull<VMFunctionBody>, NonNull<VMWasmCallFunction>>(ptr)
```

Comment:
Can this use [`NonNull::cast`](https://doc.rust-lang.org/stable/std/ptr/struct.NonNull.html#method.cast)?